### PR TITLE
Flatten input data before using stringnormalizer for a better use in pipeline

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: tests/test_sklearn_tfidf_vectorizer_converter.py [hunk 1]
```diff
@@ -29,7 +37,7 @@ def test_model_tfidf_vectorizer11(self):
         vect = TfidfVectorizer(ngram_range=(1, 1), norm=None)
         vect.fit(corpus.ravel())
         model_onnx = convert_sklearn(vect, "TfidfVectorizer",
-                                     [("input", StringTensorType([1]))],
+                                     [("input", StringTensorType())],
```
Comment (prabhat00155) [inline tests/test_sklearn_tfidf_vectorizer_converter.py]: Don't we need to pass shape?

File: skl2onnx/operator_converters/text_vectoriser.py [hunk 2]
```diff
@@ -209,6 +209,15 @@ def convert_sklearn_text_vectorizer(scope, operator, container):
             "https://github.com/onnx/sklearn-onnx/issues.")
 
     if op.lowercase or op.stop_words_:
+
```
Comment (prabhat00155) [inline skl2onnx/operator_converters/text_vectoriser.py]: What happens when the above if condition is False? flatten variable would be undefined, won't it? And that's being used below.

File: tests/test_sklearn_tfidf_vectorizer_converter.py (hunk 1 above)
Comment (xadupre) [reply tests/test_sklearn_tfidf_vectorizer_converter.py]: Not always. My goal is to test batch predictions so I need to remove that information.

File: skl2onnx/operator_converters/text_vectoriser.py (hunk 2 above)
Comment (xadupre) [reply skl2onnx/operator_converters/text_vectoriser.py]: StringNormalizer is the only operator which does not accept 2D inputs (or inputs with more than one row). Flatten is not needed when this operator is not used.

Comment (prabhat00155) [reply skl2onnx/operator_converters/text_vectoriser.py]: Check line 242, you are using flatten there(which would be undefined in the case I mentioned).

**Pull request merged**